test/calibration/test_base_calibration_experiment.py

```diff
@@ -51,3 +55,182 @@ class WrongOrder(QubitSpectroscopy, BaseCalibrationExperiment):
                 def __init__(self):
                     """A dummy class for parent order testing."""
                     super().__init__(Calibrations(coupling_map=[]), [0], [0, 1, 2])
+
+    def test_update_calibration(self):
+        """Test updating calibrations with execution of calibration experiment."""
+        backend = FakeBackend()
+        ref_new_value = 0.3
+
+        param = Parameter("to_calibrate")
+        schedule = ScheduleBlock(name="test")
+        schedule.append(Play(Constant(100, param), DriveChannel(0)), inplace=True)
+        cals = Calibrations()
+        cals.add_schedule(schedule, 0, 1)
+
+        exp = MockCalExperiment(
+            physical_qubits=(0,),
+            calibrations=cals,
+            new_value=ref_new_value,
+            param_name="to_calibrate",
+            sched_name="test",
+        )
+        exp.run(backend).block_for_results()
+
+        new_value = cals.get_parameter_value("to_calibrate", (0,), "test")
+        self.assertEqual(new_value, ref_new_value)
+
+        new_schedule = cals.get_schedule("test", (0,))
+        ref_schedule = schedule.assign_parameters({param: ref_new_value}, inplace=False)
+        self.assertEqual(new_schedule, ref_schedule)
+
+    def test_update_calibration_update_analysis(self):
+        """Test updating calibrations with experiment with updated analysis option."""
+        backend = FakeBackend()
+        ref_new_value = 0.3
+
+        param = Parameter("to_calibrate")
+        schedule = ScheduleBlock(name="test")
+        schedule.append(Play(Constant(100, param), DriveChannel(0)), inplace=True)
+        cals = Calibrations()
+        cals.add_schedule(schedule, 0, 1)
+
+        exp = MockCalExperiment(
+            physical_qubits=(0,),
+            calibrations=cals,
+            new_value=999999,
+            param_name="to_calibrate",
+            sched_name="test",
+        )
+        exp.analysis.set_options(return_value=ref_new_value)
+        exp.run(backend).block_for_results()
```

Comment:
Can you add a test that checks that the parameter obtained from the cals has a different value before the experiment? Like that we can see that the update changed the parameter.

Reply:
Note that this test already verifies the updated parameter by getting schedule from the cal instance. There is no parameter to compare because I didn't provide any initial parameter for the cal table. I added another test with initial parameter [a6dc32d](https://github.com/Qiskit/qiskit-experiments/pull/1033/commits/a6dc32dbeb3196db32f12210bf5e2319b9d82816)